src/folding/nova/ivc.rs

```diff
@@ -0,0 +1,272 @@
+use ark_crypto_primitives::sponge::{poseidon::PoseidonConfig, Absorb};
+use ark_ec::{CurveGroup, Group};
+use ark_ff::PrimeField;
+use ark_relations::r1cs::ConstraintSynthesizer;
+use ark_relations::r1cs::ConstraintSystem;
+use ark_std::rand::Rng;
+use ark_std::{One, Zero};
+use core::marker::PhantomData;
+
+use super::circuits::{AugmentedFCircuit, FCircuit};
+use super::{
+    nifs::NIFS,
+    traits::{NovaR1CS, NovaTranscript},
+    CommittedInstance, Witness,
+};
+use crate::ccs::r1cs::R1CS;
+use crate::frontend::arkworks::{extract_r1cs, extract_z}; // TODO once Frontend trait is ready, use that
+use crate::pedersen::{Params as PedersenParams, Pedersen};
+use crate::transcript::Transcript;
+use crate::{unwrap_or_return_err, Error};
+
+pub struct IVC<C1, C2, FC, Tr>
+where
+    C1: CurveGroup,
+    C2: CurveGroup,
+    FC: FCircuit<C1::ScalarField>,
+    Tr: Transcript<C1> + NovaTranscript<C1>,
+{
+    _c2: PhantomData<C2>,
+    r1cs: R1CS<C1::ScalarField>,
+    pub poseidon_config: PoseidonConfig<C1::ScalarField>,
+    pub pedersen_params: PedersenParams<C1>,
+    pub F: FC, // F circuit
+    pub transcript: Tr,
```

Comment:
Why do we need these to be `pub`? Can't we make getters and leave them private or `pub(crate)` as much to signal that they shouldn't be used?

Reply:
Right, made them not `pub`, and in fact they are not needed to be getted outside of IVC methods, so we won't need the getters at the end. Thanks for pointing it out!